client/dashboard/sites/site-change-address-modal/content.tsx

```diff
@@ -0,0 +1,346 @@
+	const { id, getValue } = field;
+	return (
+		<SuffixInputControl
+			label={ field.label }
+			value={ getValue( { item: data } ) }
+			__next40pxDefaultSize
+			suffix={ <Text variant="muted">{ DOMAIN_SUFFIX }</Text> }
+			onChange={ ( newValue ) => {
+				onChange( { [ id ]: newValue } );
+			} }
+		/>
+	);
+};
+
+const NewSiteAddressForm = ( {
+	site,
+	currentSubdomain,
+	onSubmit,
+	onCancel,
+}: {
+	site: Site;
+	currentSubdomain: string;
+	onSubmit: ( data: NewSiteAddressFormData ) => void;
+	onCancel: () => void;
+} ) => {
+	const { createErrorNotice } = useDispatch( noticesStore );
+	const { data: siteDomains } = useQuery( siteDomainsQuery( site.ID ) );
+	const mutation = useMutation( validateSiteAddressChangeMutation() );
+	const [ formData, setFormData ] = useState( {
+		subdomain: '',
+	} );
+
+	const fields: Field< NewSiteAddressFormData >[] = [
+		{
+			id: 'subdomain',
+			label: __( 'New site address' ),
+			type: 'text' as const,
+			placeholder: currentSubdomain,
+			Edit: SubdomainInputControlEdit,
+		},
+	];
+
+	const form = {
+		type: 'regular' as const,
+		fields: [ 'subdomain' ],
+	};
+
+	const hasCustomDomain = !! siteDomains?.find( ( domain ) => ! domain.wpcom_domain );
+
+	const isDisabled = ! formData.subdomain || formData.subdomain === currentSubdomain;
+
+	const handleSubmit = ( event: React.FormEvent ) => {
+		event.preventDefault();
+
+		mutation.mutate(
+			{
+				siteId: site.ID,
+				blogname: formData.subdomain,
+				domainSuffix: DOMAIN_SUFFIX.substring( 1 ),
+				siteAddressType: FreeSiteAddressType.BLOG,
+			},
+			{
+				onSuccess: () => {
+					onSubmit( formData );
+				},
+				onError: ( error ) => {
+					// TODO: Show the error via Data Form when the ValidatedInputControl is ready.
+					createErrorNotice( error.message ?? __( 'Sorry, that site address is unavailable.' ), {
+						type: 'snackbar',
+					} );
+				},
+			}
+		);
+	};
+
+	return (
+		<VStack spacing={ 4 }>
+			{ !! siteDomains && ! hasCustomDomain && (
+				<Notice variant="warning" density="medium">
+					<Text>
+						{ createInterpolateElement(
+							__(
+								'Before changing site address, consider <link>add a custom domain</link> instead?'
+							),
+							{
+								link: <ExternalLink href={ `/domains/add/${ site.slug }` } children={ null } />,
+							}
+						) }
+					</Text>
+				</Notice>
+			) }
+			<Text as="p">
+				{ sprintf(
+					/* translators: %s: site domain */
+					__( 'Once you change your site address, %s will no longer be available.' ),
+					site.slug
+				) }
+			</Text>
+			<form onSubmit={ handleSubmit }>
+				<VStack spacing={ 4 }>
+					<DataForm< NewSiteAddressFormData >
+						data={ formData }
+						fields={ fields }
+						form={ form }
+						onChange={ ( edits: Partial< NewSiteAddressFormData > ) => {
+							setFormData( ( data ) => ( { ...data, ...edits } ) );
+						} }
+					/>
+					<HStack justify="flex-end">
+						<Button variant="tertiary" disabled={ false } onClick={ onCancel }>
```

Comment:
Question: Why do we need disable={ false } instead of just not passing the disabled prop.